node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -263,8 +275,19 @@ getWalletTxHistory cWalId = toList <<$>> preview (wsTxHistory . ix cWalId)
 getWalletUtxo :: Query Utxo
 getWalletUtxo = view wsUtxo
 
+getWalletBalancesAndUtxo :: Query (WalletBalances, Utxo)
+getWalletBalancesAndUtxo = (,) <$> view wsBalances <*> view wsUtxo
```

Comment:
I will take the bullet of saying something stupid here 😀  When doing such queries, am I right in thinking these work because `Query` derives an instance of `MonadReader`? If so, are we sure that using the applicative form here won't result into 2 DB accesses? Aka I'm wondering if doing something like the following would be equivalent (or not) in terms of DB access:

``` haskell
getWalletBalancesAndUtxo = do
  db <- ask
  (view wsBalances db, view wsUtxo db)
```

In case I'm talking gibberish, please ignore 😂

Reply:
AFAIK acid-state store db in memory and two there `view` equals two reads from Reader.